### PR TITLE
PLNSRVCE-1431: fix time to substract amount for gap metric

### PR DESCRIPTION
**Pull request by gabemontero**
@enarha @ramessesii2 so https://github.com/openshift-pipelines/pipeline-service-exporter/pull/34 addressed the negative metrics, but there was a slight bug that resulted in values being too big

this addresses that, where I pulled in more staging yaml to first reproduce and then validate the fix

@jhutar FYI

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -303,6 +303,7 @@ func (c *PipelineRunTaskRunGapCollector) bumpGapDuration(pr *v1beta1.PipelineRun
 			if !tr2.Status.CompletionTime.Time.After(tr.CreationTimestamp.Time) {
 				ctrl.Log.V(4).Info(fmt.Sprintf("%s did not complete after so use it to compute gap for current task %s", taskRunTaskRef(tr2, pr), taskRunTaskRef(tr, pr)))
 				trToCalculateWith = tr2
+				timeToCalculateWith = tr2.Status.CompletionTime.Time
```
Comment (gabemontero) [inline collector/collector.go:306]: this is the crux of the fix

the rest is added testing, plus some cleanup

File: collector/collector_test.go [hunk 2]
```diff
@@ -360,6 +362,9 @@ func validateHistogramVec(t *testing.T, h *prometheus.HistogramVec, labels prome
 	assert.NotZero(t, *metric.Histogram.SampleCount)
 	assert.NotNil(t, metric.Histogram.SampleSum)
 	assert.Greater(t, *metric.Histogram.SampleSum, float64(-1))
+	if checkMax {
+		assert.Less(t, *metric.Histogram.SampleSum, float64(7001))
```
Comment (enarha) [inline collector/collector_test.go:366]: Where the 7001 limit comes from?

**Review by enarha — COMMENTED** [review]
/lgtm

**Review by ramessesii2 — COMMENTED** [review]
/lgtm

File: collector/collector_test.go [hunk 3]
```diff
@@ -374,22 +379,33 @@ func validateGaugeVec(t *testing.T, g *prometheus.GaugeVec, labels prometheus.La
...
+	return prs, nil
 }
 
 func taskRunsFromActualRHTAPYaml() ([]v1beta1.TaskRun, error) {
 	trs := []v1beta1.TaskRun{}
-	yamlStrings := []string{trInitYaml,
+	yamlStrings := []string{tooBigNumTRInitYaml,
+		tooBigNumTRCloneYaml,
+		tooBigNumTRBuildYaml,
+		tooBigNumTRSbonYaml,
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply collector/collector_test.go:406]: ack

**Review by enarha — DISMISSED** [review]

**Pull request merged**